emailer.py

```diff
@@ -70,6 +76,73 @@ class Emailer(object):
         'from_address',
     ]
 
+    ##### Public Interface / Magic Methods ###################################  # noqa: E266
+    def __init__(self, smtp_username, smtp_password, smtp_host, smtp_port, from_name, from_address, templates):
+        config_errors = []
+        required_parameters = ('smtp_username', 'smtp_password', 'smtp_host', 'smtp_port', 'from_name', 'from_address')
+        for param_name in required_parameters:
+            if not locals()[param_name]:
+                config_errors.append(param_name)
+
+        if config_errors:
+            msg = "Emailer instantiated with missing params: " + ", ".join(config_errors)
+            raise CannotLoadConfiguration(msg)
+
+        self.smtp_username = smtp_username
+        self.smtp_password = smtp_password
+        self.smtp_host = smtp_host
+        self.smtp_port = smtp_port
+        self.from_name = from_name
+        self.from_address = from_address
+        self.templates = templates
+
+        # Make sure the templates don't contain any template values we can't handle.
+        test_template_values = dict(
+            (key, "value") for key in self.KNOWN_TEMPLATE_KEYS
+        )
+        for template in list(self.templates.values()):
+            try:
+                template.body("from address", "to address", **test_template_values)
+            except Exception as e:
+                m = f"Template '{template.subject_template}'/'{template.body_template}' contains unrecognized key: {e}"
+                raise CannotLoadConfiguration(m)
+
+    def send(self, email_type, to_address, smtp=None, **kwargs):
+        """Generate an email from a template and send it.
+
+        :param email_type: The name of the template to use.
+        :param to_address: Addressee of the email.
+        :param smtp: Use this object as a mock instead of creating an
+            smtplib.SMTP object.
+        :param kwargs: Arguments to use when generating the email from
+            a template.
+        """
+        if email_type not in self.templates:
+            raise ValueError("No such email template: %s" % email_type)
+        template = self.templates[email_type]
+        from_header = '%s <%s>' % (self.from_name, self.from_address)
+        kwargs['from_address'] = self.from_address
+        kwargs['to_address'] = to_address
+        body = template.body(from_header, to_address, **kwargs)
+
+        try:
+            self._send_email(to_address, body, smtp)
+        except Exception as exc:
+            raise CannotSendEmail(exc)
```

Comment:
Attempt to actually send the email. If a generic exception comes through, raise `CannotSendEmail`.